web_timeline/static/src/js/web_timeline.js

```diff
@@ -0,0 +1,514 @@
+            _.each(events, function(event) {
+                if (event[self.date_start]){
+                    data.push(self.event_data_transform(event));
+                }
+            });
+            // get the groups
+            var split_groups = function(events, group_bys) {
+                if (group_bys.length === 0)
+                    return events;
+                var groups = [];
+                groups.push({id:-1, content: _t('-')})
+                _.each(events, function(event) {
+                    var group_name = event[_.first(group_bys)];
+                    if (group_name) {
+                        var group = _.find(groups, function(group) { return _.isEqual(group.id, group_name[0]); });
+                        if (group === undefined) {
+                            group = {id: group_name[0], content: group_name[1]};
+                            groups.push(group);
+                        }
+                    }
+                });
+                return groups;
+            }
+            var groups = split_groups(events, group_bys);
+            this.timeline.setGroups(groups);
+            this.timeline.setItems(data);
+            this.timeline.fit();
+        },
+
+        do_show: function() {
+            this.do_push_state({});
+            return this._super();
+        },
+
+        is_action_enabled: function(action) {
+            if (action === 'create' && !this.options.creatable) {
+                return false;
+            }
+            return this._super(action);
+        },
+
+        on_add: function(item, callback) {
+            var self = this;
+            var context = this.dataset.get_context();
+            // Initialize default values for creation
+            var default_context = {}
+            default_context['default_'.concat(this.date_start)] = item.start;
+            default_context['default_'.concat(this.date_stop)] = moment(item.start).add(1, 'hours').toDate();
+            if (item.group > 0) {
+                default_context['default_'.concat(this.last_group_bys[0])] = item.group;
+            }
+            context.add(default_context);
+            // Show popup
+            var dialog = new form_common.FormViewDialog(this, {
+                res_model: this.dataset.model,
+                res_id: null,
+                context: context,
+                view_id: +this.open_popup_action,
+            }).open();
+            dialog.on('create_completed', self, function(id) {
+                self.dataset.ids = this.dataset.ids.concat([id]);
+                self.dataset.trigger("dataset_changed", id);
+                self.dataset.read_ids([id], self.fields).done(function(records) {
+                    var new_event = self.event_data_transform(records[0]);
+                    var items = self.timeline.itemsData;
+                    items.add(new_event);
+                    self.timeline.setItems(items);
+                });
+            });
```

Comment:
Again, you can define a `on_create_completed` method in the object and call it here with a proxy, to let it be inherited/extended.